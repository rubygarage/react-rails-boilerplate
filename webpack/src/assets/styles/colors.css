body {
  background-color: #EFEFEF;
}

.primary-color {
  background: rgb(96,125,139);
}

.secondary-color {
  background: rgb(255,82,82);
}
