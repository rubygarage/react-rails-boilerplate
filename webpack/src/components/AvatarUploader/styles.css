.avatar-preview-wrap {
  height: 100%;
  width: 100%;
  position: relative;
}

.avatar-uploader-thumb {
  height: 100%;
  width: 100%;
  object-fit: cover;
}

.remove-avatar-preview-button {
  position:absolute;
  top: 5px;
  right: 5px;
  font-size: 13px;
  height: 20px;
  width: 20px;
  border-radius: 10px;
  border: none;
  color: #ffffff;
  background-color: #333333;
  cursor: pointer;
}
