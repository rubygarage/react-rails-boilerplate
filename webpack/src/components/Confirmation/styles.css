.content {
  display: flex;
  flex-grow: 1;
  align-items: center;
  justify-content: center;
  padding-bottom: 13.5rem;
  max-width: 60rem;
  margin-left: auto;
  margin-right: auto;
}
