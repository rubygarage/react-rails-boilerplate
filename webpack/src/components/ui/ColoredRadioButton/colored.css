.theme-color {
  height: 3.6rem;
  width: 3.6rem;
  border-radius: 0.3rem;
  margin-right: 1.5rem;
  margin-bottom: 1.5rem;
  display: flex;
  justify-content: center;
  align-items: center;

  &:hover {
    cursor: pointer;
  }
}

.radio-input {
  display: none;
}

.radio-checked {
  border: solid 0.1rem white;

  .radio-icon:before {
    content: "\e6a2";
  }
}

.pink {
  background-color: #d24d57;
}

.red {
  background-color: #c04930;
}

.orange {
  background-color: #ba5d03;
}

.brown {
  background-color: #8f704e;
}

.olive {
  background-color: #677841;
}

.green {
  background-color: #117454;
}

.blue {
  background-color: #0086bc;
}

.ocean {
  background-color: #004d65;
}

.indigo {
  background-color: #7e5a9d;
}

.purple {
  background-color: #53354e;
}

.grey {
  background-color: #404040;
}

.black {
  background-color: #0d0d0d;
}

@media (min-width: 1200px) {
  .theme-color {
    height: 2.3rem;
    width: 2.3rem;
  }
}
