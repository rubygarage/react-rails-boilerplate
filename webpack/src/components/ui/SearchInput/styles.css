.label-search {
  position: relative;
  display: inline-block;
  margin-bottom: 0;
  max-width: 100%;
}

.input-search {
  padding: 0.5rem 1rem 0.5rem 4rem;

  &:focus {
    padding: 0.5rem 1rem 0.5rem 4rem;
  }
  &.disable {
    color: #6d6d6d;
    cursor: not-allowed;
  }
}

.search-icon {
  position: absolute;
  left: 1rem;
  top: 50%;
  transform: translateY(-50%);
  font-size: 2rem;
  color: #ccc;
  opacity: 0.3;
}
