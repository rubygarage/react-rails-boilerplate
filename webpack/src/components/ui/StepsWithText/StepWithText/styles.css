@media (min-width: 768px) {
  .step {
    min-width: 10rem;
    display: inline-block;
    font-size: 1.2rem;
    padding: 0.5rem 0.3rem;
    color: #6d6d6d;
    position: relative;

    &.active {
      color: white;
    }

    &:before {
      content: '';
      height: 0.7rem;
      width: 0.7rem;
      border-radius: 50%;
      background-color: currentColor;
      position: absolute;
      bottom: 0;
      left: 50%;
      transform: translate(-50%, 100%);
    }

    &:after {
      content: '';
      height: 0.1rem;
      width: 80%;
      background-color: rgba(255, 255, 255, 0.2);
      position: absolute;
      bottom: -0.3rem;
      left: 60%;
      transform: translate(0%, 100%);
    }

    &:last-of-type:after {
      content: none;
    }
  }

  .active {
    &.step {
      color: white;
    }

    .step-text {
      color: white;
    }
  }

  .passed {
    &.step {
      color: #637fa0;
    }

    .step-text {
      color: #6d6d6d;
    }
  }
}

@media (min-width: 1200px) {
  .step {
    min-width: 11rem;
  }
}
