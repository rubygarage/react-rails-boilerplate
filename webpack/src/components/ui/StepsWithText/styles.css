@media (min-width: 768px) {
  .menu {
    display: flex;
    justify-content: center;
  }
}
