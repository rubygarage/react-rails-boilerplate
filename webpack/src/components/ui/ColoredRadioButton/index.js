import React, { Component } from 'react'
import classNames from 'classnames'
import { Field } from 'redux-form'

import styles from './colored.css'

class ColoredRadioButton extends Component {
  field = ({ input, className }) => (
    <label className={classNames('radio', className, { [styles['radio-checked']]: input.checked })}>
      <input {...input} type="radio" className={classNames(styles['radio-input'])} />
      <span className={classNames(styles['radio-icon'], 'icon')} />
    </label>
  )

  render() {
    return <Field {...this.props} type="radio" component={this.field} />
  }
}

export default ColoredRadioButton
