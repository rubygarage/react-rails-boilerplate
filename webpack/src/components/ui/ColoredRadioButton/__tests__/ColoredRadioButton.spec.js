import React from 'react'
import { shallow } from 'enzyme'
import classNames from 'classnames'
import ColoredRadioButton from 'components/ui/ColoredRadioButton'

import styles from '../colored.css'

describe('<ColoredRadioButton />', () => {
  const defaultProps = {
    className: classNames(styles['theme-color'], styles.black)
  }

  it('renders with empty props', () => {
    const component = shallow(<ColoredRadioButton />)

    expect(component).toMatchSnapshot()
  })

  it('renders with props', () => {
    const component = shallow(<ColoredRadioButton {...defaultProps} />)

    expect(component).toMatchSnapshot()
  })
})
