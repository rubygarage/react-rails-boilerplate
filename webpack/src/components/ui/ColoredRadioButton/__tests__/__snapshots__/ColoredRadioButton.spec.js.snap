// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ColoredRadioButton /> renders with empty props 1`] = `
<Field
  component={[Function]}
  type="radio"
/>
`;

exports[`<ColoredRadioButton /> renders with props 1`] = `
<Field
  className="theme-color black"
  component={[Function]}
  type="radio"
/>
`;
